apps/aecore/src/aec_headers.erl

```diff
@@ -38,24 +36,55 @@ time_in_secs(Header) ->
     Time = Header#header.time,
     aeu_time:msecs_to_secs(Time).
 
--spec serialize_for_network(header()) ->
-                                   {ok, block_header_serialized_for_network()}.
-serialize_for_network(H = #header{}) ->
-    %% TODO: Define serialization format.
-    {ok, term_to_binary(H)}.
-
--spec deserialize_from_network(block_header_serialized_for_network()) ->
-                                      {ok, header()}.
-deserialize_from_network(H) when is_binary(H) ->
-    %% TODO: Define serialization format.
-    {ok, #header{} = binary_to_term(H)}.
-
--spec hash_network_serialization(block_header_serialized_for_network()) ->
-                                        {ok, block_header_hash()}.
-hash_network_serialization(H) when is_binary(H) ->
-    {ok, aec_sha256:hash(H)}.
-
--spec hash_internal_representation(header()) -> {ok, block_header_hash()}.
-hash_internal_representation(H = #header{}) ->
-    {ok, HH} = serialize_for_network(H),
-    hash_network_serialization(HH).
+-spec serialize_to_map(header()) -> {ok, map()}.
+serialize_to_map(H = #header{}) ->
+    Serialized =
+      #{<<"height">> =>  height(H),
+        <<"prev-hash">> => base64:encode(prev_hash(H)),
+        <<"root-hash">> => base64:encode(H#header.root_hash),
+        <<"target">> => H#header.target,
+        <<"nonce">> => H#header.nonce,
+        <<"time">> => H#header.time,
+        <<"version">> => H#header.version
+      },
+    {ok, Serialized}.
+
+-spec deserialize_from_map(map()) -> {ok, header()}.
+deserialize_from_map(H = #{}) ->
+      #{<<"height">> := Height,
+        <<"prev-hash">> := PrevHash,
+        <<"root-hash">> := RootHash,
+        <<"target">> := Target,
+        <<"nonce">> := Nonce,
+        <<"time">> := Time,
+        <<"version">> := Version 
+      } = H,
+    {ok, #header{height = Height,
+                 prev_hash = base64:decode(PrevHash),
+                 root_hash = base64:decode(RootHash),
+                 target = Target,
+                 nonce = Nonce,
+                 time = Time,
+                 version = Version}}.
+
+-spec serialize_to_binary(header()) -> {ok, header_binary()}.
+serialize_to_binary(H) ->
+    {ok, Map} = serialize_to_map(H),
+    {ok, jsx:encode(Map)}.
+    %<<(H#header.height):64,
+    %  (H#header.prev_hash):(?BLOCK_HEADER_HASH_BYTES*8),
+    %  (H#header.root_hash):(?BLOCK_HEADER_HASH_BYTES*8),
+    %  (H#header.target):64,
+    %  (H#header.nonce):64,
+    %  (H#header.time):64,
+    %  (H#header.version):16>>.
+
+-spec deserialize_from_binary(header_binary()) -> {ok, header()}.
+deserialize_from_binary(B) ->
+    deserialize_from_map(jsx:decode(B, [return_maps])).
+
+-spec hash_header(header()) -> {ok, block_header_hash()}.
+hash_header(H) ->
```

Comment:
Please add TODO comment mentioning that this hash is meant to be deterministic and based on the canonical serialization of the header.